staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1048,77 +1049,86 @@ func seedMultiLevelData(ctx context.Context, store storage.Interface) (string, [
+		expectedOut: []example.Pod{*storedObj},
 	}, {
-		name:        "existing key, resourceVersion=0",
-		key:         key,
-		pred:        storage.Everything,
-		expectedOut: []*example.Pod{storedObj},
-		rv:          "0",
+		name:                 "existing key, resourceVersion=0",
+		key:                  key,
+		pred:                 storage.Everything,
+		expectedAlternatives: [][]example.Pod{nil, {*storedObj}},
+		rv:                   "0",
 	}, {
-		name:        "existing key, resourceVersion=0, resourceVersionMatch=notOlderThan",
-		key:         key,
-		pred:        storage.Everything,
-		expectedOut: []*example.Pod{storedObj},
-		rv:          "0",
-		rvMatch:     metav1.ResourceVersionMatchNotOlderThan,
+		name:                 "existing key, resourceVersion=0, resourceVersionMatch=notOlderThan",
+		key:                  key,
+		pred:                 storage.Everything,
+		expectedAlternatives: [][]example.Pod{nil, {*storedObj}},
+		rv:                   "0",
+		rvMatch:              metav1.ResourceVersionMatchNotOlderThan,
 	}, {
-		name:        "existing key, resourceVersion=current",
-		key:         key,
-		pred:        storage.Everything,
-		expectedOut: []*example.Pod{storedObj},
-		rv:          fmt.Sprintf("%d", currentRV),
+		name:                 "existing key, resourceVersion=previous, resourceVersionMatch=notOlderThan",
+		key:                  key,
+		pred:                 storage.Everything,
+		expectedAlternatives: [][]example.Pod{{*prevStoredObj}, {*storedObj}},
+		rv:                   fmt.Sprintf("%d", prevRV),
+		rvMatch:              metav1.ResourceVersionMatchNotOlderThan,
 	}, {
 		name:        "existing key, resourceVersion=current, resourceVersionMatch=notOlderThan",
 		key:         key,
 		pred:        storage.Everything,
-		expectedOut: []*example.Pod{storedObj},
+		expectedOut: []example.Pod{*storedObj},
 		rv:          fmt.Sprintf("%d", currentRV),
 		rvMatch:     metav1.ResourceVersionMatchNotOlderThan,
 	}, {
-		name:        "existing key, resourceVersion=previous, resourceVersionMatch=notOlderThan",
+		name:        "existing key, resourceVersion=current",
 		key:         key,
 		pred:        storage.Everything,
-		expectedOut: []*example.Pod{storedObj},
-		rv:          fmt.Sprintf("%d", prevRV),
-		rvMatch:     metav1.ResourceVersionMatchNotOlderThan,
+		expectedOut: []example.Pod{*storedObj},
+		rv:          fmt.Sprintf("%d", currentRV),
 	}, {
 		name:        "existing key, resourceVersion=current, resourceVersionMatch=exact",
 		key:         key,
 		pred:        storage.Everything,
-		expectedOut: []*example.Pod{storedObj},
+		expectedOut: []example.Pod{*storedObj},
 		rv:          fmt.Sprintf("%d", currentRV),
 		rvMatch:     metav1.ResourceVersionMatchExact,
 	}, {
-		name:        "existing key, resourceVersion=current, resourceVersionMatch=exact",
-		key:         prevKey,
+		name:        "existing key, resourceVersion=previous, resourceVersionMatch=exact",
+		key:         key,
```

Comment:
key and prevKey were the same (before I broke it in [#113427](https://github.com/kubernetes/kubernetes/pull/113427)) and I fixed that again